base/arrayshow.jl

```diff
@@ -423,6 +422,12 @@ function show(io::IO, X::AbstractArray)
         _show_nonempty(io, X, prefix)
 end
 
+### 0-dimensional arrays -- see https://github.com/JuliaLang/julia/issues/31481
+function show_zero_dim(io::IO, X::AbstractArray{<:Any, 0})
+    val = isassigned(X) ? repr(X[]) : undef_ref_str
+    print(io, "fill($val)")
```

Comment:
`X[]` should be shown to the `io` stream instead of being stringified separately.

Reply:
Thanks!

To be clear (i'm still getting my head round printing stuff)
the change is just dropping the `repr` call, i.e.:
```julia
val = isassigned(X) ? X[] : undef_ref_str
print(io, "fill($val)")
```

Reply:
It should be something like
```
print(io, "fill(")
show(io, X[])
...
```
So that the `X[]` value can "see" any context settings in `io`.

Reply:
ah, that makes sense! thank you!

Reply:
I made a change, passing `context=io` to `repr`

becuase with
```julia
print(io, "fill(")
show(io, isassigned(X) ? X[] : undef_ref_str)
print(io, ")")
```
would show `Array{Any,0}(undef)` as `fill("#undef")` when (I think?) it should be `fill(#undef)`

But perhaps there is a better way, or I am wrong about the result we want?

Reply:
`isassigned(X) ? show(io, X[]) : print(io, undef_ref_str) `

Reply:
Thanks! Changed it to that

If someone could let me know how/why that's preferable to `repr(x, context=io)` (or point me to the right docs), that might help me in future -- sorry to be learning printing stuff on the fly here :)

Reply:
Both work; making intermediate strings just seems inefficient.